src/basic/terminal-util.c

```diff
@@ -253,8 +253,9 @@ int reset_terminal_fd(int fd, bool switch_to_text) {
         if (switch_to_text)
                 (void) ioctl(fd, KDSETMODE, KD_TEXT);
 
-        /* Set default keyboard mode */
-        (void) vt_reset_keyboard(fd);
+        /* Set default keyboard mode without breaking it */
+        if (vt_verify_kbmode(fd) == 0)
```

Comment:
not that it matetrs much, but we usually check for success with >= 0, as for most of our functions return values < 0 are failure, and >= 0 are success.

Reply:
Sure, done.